### PR TITLE
pyside2: Add qt3d to buildInputs

### LINKED ISSUE #191313: PySide2 missing Qt3DCore module
### Describe the bug
The python3 package `pyside2` appears to be missing the `Qt3DCore` submodule.

### Steps To Reproduce
1. `$ nix-shell -p 'python3.withPackages (ps: with ps; [pyside2])' --run python3`
2. `>>> import PySide2.Qt3DCore`
```console
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'PySide2.Qt3DCore'
```

### Expected behavior
The python `import PySide2.Qt3DCore` statement to succeed.

### Additional Context
https://doc.qt.io/qtforpython-5/PySide2/Qt3DCore/index.html

### Notify maintainers
@gebner 

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.63, NixOS, 22.05 (Quokka), 22.05.20220831.067d5d5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(aidan): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by hesiod**
###### Description of changes

PySide2 will not build its qt3d component unless qt3d is added to buildInputs.
This is part of the effort to package Meshroom (https://github.com/NixOS/nixpkgs/issues/94127).
(Related Meshroom PRs: #215528 and #215728)

Fixes #191313

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hesiod** [comment]
Result of `nixpkgs-review pr 215729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>tumpa</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>napari (python310Packages.napari)</li>
    <li>python311Packages.magicgui</li>
    <li>python311Packages.napari-npe2</li>
    <li>python311Packages.pyside2</li>
    <li>python311Packages.pyside2-tools</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>cutter</li>
    <li>freecad</li>
    <li>hydrus</li>
    <li>onionshare-gui</li>
    <li>patray</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.pyside2</li>
    <li>python310Packages.pyside2-tools</li>
    <li>renderdoc</li>
    <li>sl1-to-photon</li>
    <li>streamdeck-ui</li>
    <li>syncplay</li>
  </ul>
</details>

**Comment by hesiod** [comment]
Regarding the build failures:
* `python311Packages.pyside2` and friends were never supported because [upstream does not support Python > 3.10](https://bugreports.qt.io/browse/PYSIDE-1864). I've taken the liberty of disabling pyside2 for Python 3.11 and later. It looks like the Red Hat folks backported some patches (see [the bug report](https://bugzilla.redhat.com/show_bug.cgi?id=2025599), but I'm not sure whether that's worth the hassle).
* `python310Packages.napari` was previously broken (upstream requires `napari-svg>=0.1.6`, but we only have `0.1.5` at the moment

In summary, there are no new build failures due to this PR.

**Comment by hesiod** [comment]
Result of `nixpkgs-review pr 215729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>tumpa</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>napari (python310Packages.napari)</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>cutter</li>
    <li>freecad</li>
    <li>hydrus</li>
    <li>onionshare-gui</li>
    <li>patray</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.pyside2</li>
    <li>python310Packages.pyside2-tools</li>
    <li>renderdoc</li>
    <li>sl1-to-photon</li>
    <li>streamdeck-ui</li>
    <li>syncplay</li>
  </ul>
</details>

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, nixos x86_64

**Comment by davidak** [comment]
>[upstream does not support Python > 3.10](https://bugreports.qt.io/browse/PYSIDE-1864).

we don't have to be better than upstream, so that is fine

as i understand, they will support python 3.11 in later versions. so this seems not an issue

**Review by davidak — APPROVED** [review]
Change looks good.
